### DISCUSSION TITLE
pt_head shows exact zero initial loss with true DFT replay data (Method 1), MACE 0.3.16

### DISCUSSION THREAD

**Discussion by zbschwab** [ACEsuit/mace#1527] [Q&A]
### Setup
* MACE version: 0.3.16
* Foundation model: mace-mh-1, foundation_head: omat_pbe
* Replay method: Method 1 (fine_tuning_select.py), true DFT labels 
* Target dataset (`train.xyz`): 318 frames (159 labeled first/last-frame pairs) of my own DFT runs
* Replay dataset (`selected_configs.xyz`): 811 frames, this is the max `fine_tuning_select.py` returns after filtering the released replay file to my elements [1, 6, 8, 47]. From `replay-data-mh-1-omat-pbe.xyz` source file.

### Commands used
**Selection:**
```
python -m mace.cli.fine_tuning_select \
  --configs_pt replay-data-mh-1-omat-pbe.xyz \
  --configs_ft train.xyz \
  --num_samples 811 \
  --subselect fps \
  --model mace-mh-1.model \
  --output selected_configs.xyz \
  --filtering_type combinations \
  --filter_atomic_numbers_pt "[1, 6, 8, 47]" \
  --head_pt omat_pbe \
  --head_ft dft_ag \
  --weight_pt 1.0 \
  --weight_ft 10.0 \
  --default_dtype float32 \
  --disallow_random_padding
```

**Training:**
```
python -m mace.cli.run_train \
--name Ag_CHO_finetuned_mace-mh-1 \
--train_file data/train.xyz \
--foundation_model mh-1 \
--foundation_head omat_pbe \
--pt_train_file data/selected_configs.xyz \
--energy_weight 1.0 --forces_weight 100.0 \
--swa --swa_energy_weight 10.0 --swa_forces_weight 100.0 \
--stress_weight 0.0 \
--lr 0.0001 --swa_lr 0.00001 \
--multiheads_finetuning True \
--valid_fraction 0.1 \
--E0s "{47: -0.19458509, 8: -1.55774442, 6: -1.28369203, 1: -1.11167391}" \
--max_num_epochs 30 --device cuda
```

### Issue
The log file shows `pt_head` is exactly zero (`loss=0.00000000`, RMSE_E/F/stress all `0.00`) at the Initial evaluation, before any gradient update. I would expect low but not bitwise-exact zero error here. 
```
Initial: head: pt_head, loss=0.00000000, RMSE_E_per_atom=0.00 meV, RMSE_F=0.00 meV/A, RMSE_stress=0.00
Initial: head: Default, loss=0.00557605, RMSE_E_per_atom=432.74 meV, RMSE_F=67.09 meV/A, RMSE_stress=None
```
I've checked that `selected_configs.xyz` and `train.xyz` are written correctly/have actual REF values. Loss rises past zero at Epoch 0 and spikes at Epoch 1, so the issue seems to affect the initial pass. 

Overall training results for this run look far worse than expected, and this exact-zero Initial reading is the only concrete irregularity I've been able to pin down after checking data integrity and config flags, so I'm not sure whether it indicates the actual cause.
Is this zero reading a cosmetic logging issue, or does it indicate pt_head's data isn't actually being used correctly during training? 

Any suggestions on what could be going wrong or how to verify pt_head is training on real labels would be appreciated. Full log file attached.

[mace_732830.txt](https://github.com/user-attachments/files/30144618/mace_732830.txt)

**Comment by CoderDeltaLAN** [comment]
Hi from the log the replay data is being used: 730 `pt_head` frames are loaded, and its loss becomes non-zero from epoch 0. The exact-zero initial line is still odd for Method 1, so I’d compare one replay frame against the untouched `omat_pbe` head. Also, `--weight_ft 10.0` won’t affect the original `train.xyz`, and SWA turns stress back on unless you add `--swa_stress_weight 0.0`. The final `pt_head` skip only affects the summary table
